G1ANT.Addon.Net/SimplifiedMessageSummary.cs

```diff
@@ -1,27 +1,321 @@
 ﻿using MailKit;
 using MimeKit;
-using G1ANT.Language;
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections.Generic;
+using G1ANT.Addon.Net.Models;
 
 namespace G1ANT.Addon.Net
 {
-    public class SimplifiedMessageSummary : MessageSummary
+    public class SimplifiedMessageSummary
     {
-        public new string TextBody;
         public IMailFolder Folder;
-        public ListStructure MessageAttachments;
 
-        public SimplifiedMessageSummary(MessageSummary summary, IMailFolder folder, ListStructure attachmentsList) : base(summary.Index)
+        private MessageSummary messageSummary = null;
+        private MimeMessage fullMessage = null;
+
+        public SimplifiedMessageSummary()
         {
-            TextBody = ParseBody(summary, folder);
-            MessageAttachments = attachmentsList;
-            Envelope = summary.Envelope;
-            UniqueId = summary.UniqueId;
+            FullMessage = new MimeMessage();
+        }
+
+        public SimplifiedMessageSummary(MessageSummary summary, IMailFolder folder)
+        {
+            messageSummary = summary;
             Folder = folder;
         }
 
-        private static string ParseBody(IMessageSummary message, IMailFolder folder)
+        protected SimplifiedMessageSummary(MimeMessage message)
+        {
+            FullMessage = message;
+        }
+
+        public MimeMessage FullMessage
+        {
+            get
+            {
+                if (fullMessage == null && messageSummary != null)
+                    fullMessage = Folder.GetMessage(messageSummary.UniqueId);
+                return fullMessage;
+            }
+            set
+            {
+                fullMessage = value;
+            }
+        }
+
+        public List<object> Attachments
+        {
+            get => CreateAttachmentStructuresFromAttachments(FullMessage.Attachments);
+            set
+            {
+                var bodyBuilder = new BodyBuilder();
+                if (bodyBuilder.Attachments != null)
+                {
+                    bodyBuilder.HtmlBody = HtmlBody;
+                    bodyBuilder.TextBody = TextBody;
+                    foreach (var attachment in value)
+                        bodyBuilder.Attachments.Add(attachment.ToString());
+                    FullMessage.Body = bodyBuilder.ToMessageBody();
+                }
+            }
+        }
+
+        private List<object> CreateAttachmentStructuresFromAttachments(IEnumerable<MimeEntity> attachments)
+        {
+            List<object> attachmentsList = new List<object>();
+
+            foreach (var attachment in attachments.Where(x => !string.IsNullOrEmpty(x.ContentDisposition?.FileName)))
```

Comment:
Consider
```csharp
var attachmentsList = GetAttachmentsWithNamesSet(attachments)
    .Select(a => new AttachmentStructure(new AttachmentModel(a)))
    .ToList();
```

Reply:
This code has not been modified when moving SimplifiedMessageSummary class from G1ANT.Language but ok 👍